### ISSUE TITLE
Local DHT?

### ISSUE THREAD

**Issue by eduhenke** [libp2p/go-libp2p#322]
tl;dr - I'm trying to create a local DHT, so I can run tests locally, but I have no idea how to do that

My program consists of
- creating libp2p-hosts:
```host, err := libp2p.New(ctx, libp2p.Defaults, libp2p.NATPortMap())```
- running a ipfs bootstrapper:
```$ docker run -d --name ipfs_host -v $ipfs_staging:/export -v $ipfs_data:/data/ipfs -p 4001:4001 -p 127.0.0.1:8080:8080 -p 127.0.0.1:5001:5001 ipfs/go-ipfs:latest```
- connecting the host to the ipfs bootstrapper:
```
iaddr, err := ipfsaddr.ParseString(addr)
pinfo, _ := peerstore.InfoFromP2pAddr(iaddr.Multiaddr())

if err := host.Connect(ctx, *pinfo); err != nil {
    fmt.Println("bootstrapping to peer failed: ", err)
}
```
- announcing(providing) a topic with a libp2p-kad-dht and a cid that were created:
`dht.Provide(timedCtx, cid, true)`
- finding and connecting to the peers:
```
peers, err := dht.FindProviders(ctx, cid)
...
host.Connect(tctx, peerId)
```
It is working all fine, but the problem is, I'm trying to remove the ipfs-bootstrappers from my program, so I can run tests locally.
I've managed to connect two peers locally, not depending from a external bootstrapper(ipfs)
but when I'm providing with the kad-dht:
`if err := dht.Provide(timedCtx, cid, true); err != nil {`
it gives me the error: `failed to find any peer in table`
I'm sure I'm doing something silly, but the main program runs fine, with the external(ipfs) bootstrapper.

**Comment by Stebalien** [comment]
Please direct questions to [discuss.libp2p.io](https://discuss.libp2p.io).
